.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,28 @@
+name: e2e
+on:
+  pull_request:
+    paths-ignore:
+    - '**/*.md'
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+jobs:
+  kind:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '~1.17'
+    - name: Install kind
+      run: |
+        curl -sLo kind "$(curl -sL https://api.github.com/repos/kubernetes-sigs/kind/releases/latest | jq -r '[.assets[] | select(.name == "kind-linux-amd64")] | first | .browser_download_url')"
+        chmod +x kind
+        sudo mv kind /bin/
+    - name: Create kind cluster
+      run: |
+        kind create cluster
+        kind export kubeconfig
```

Comment:
I don't think we need a separate kind cluster if we use kind's go packages directly [ala olm](https://github.com/operator-framework/operator-lifecycle-manager/blob/b414e6e881066115b4c5dc63840e8d47c624a405/.github/workflows/e2e-tests.yml?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L15)

[see also](https://github.com/operator-framework/operator-lifecycle-manager/blob/b414e6e881066115b4c5dc63840e8d47c624a405/test/e2e/ctx/provisioner_kind.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L91)

Reply:
Can we make this kind of improvement as a follow-up issue? Maybe this points towards pursuing a shared operator-framework/test-infra so we're not duplicating work across OLM/registry/combo/etc.?